src/App.tsx

```diff
@@ -1,223 +1,143 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useEffect, useMemo, useContext, useState, useCallback, FormEvent,
+} from 'react';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { ErrorNotification } from './components/ErrorNotification';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Footer } from './components/Footer';
+import { Todo } from './types/Todo';
+import { getTodos, addTodo, deleteTodo } from './api/todos';
+import { FilterType } from './types/Filter';
+import { ErrorMessage } from './types/Error';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
-  const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filterType, setFilterType] = useState<FilterType>(FilterType.All);
+  const [isLoading, setIsLoading] = useState(false);
+  const [title, setTitle] = useState('');
+  const [isSelectId, setIsSelectId] = useState<number[]>([]);
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
-    if (newTodoField.current) {
-      newTodoField.current.focus();
+    const getTodoFromServer = async (userId: number) => {
+      try {
+        const todosFromServer = await getTodos(userId);
+
+        setTodos(todosFromServer);
+      } catch (error) {
+        setErrorMessage(`${error}`);
+      }
+    };
+
+    if (!user) {
+      return;
     }
+
+    getTodoFromServer(user.id);
   }, []);
 
+  const filteredTodos = useMemo(() => {
+    return todos.filter((todo) => {
+      switch (filterType) {
+        case FilterType.Active:
+          return !todo.completed;
+
+        case FilterType.Completed:
+          return todo.completed;
+
+        default:
+          return todo;
+      }
+    });
+  }, [todos, filterType]);
+
+  const isActiveTodos = useMemo(() => {
+    return todos.some(todo => !todo.completed);
+  }, [todos]);
+
+  const newTodo = useCallback(async (event: FormEvent) => {
+    event.preventDefault();
+
+    if (!user || !title) {
+      setErrorMessage(ErrorMessage.ErrorTitle);
+
+      return;
+    }
+
+    setIsLoading(true);
+
+    try {
+      const addedTodo = await addTodo(title, user.id);
+
+      setTodos([...todos, addedTodo]);
```

Comment:
If the new state is computed using the previous state, you can pass a function to setTodos